send-report.sh

```diff
@@ -0,0 +1,209 @@
+author_ident_sys="$3"
+author_ident_val="$4"
+proj_ident_sys="$5"
+proj_ident_val="$6"
+
+user=""
+password=""
+issuer_url=""
+client_id=""
+client_secret=""
+
+
+read_http_args() {
+
+    while getopts ":u:p:i:c:s:" opt; do
+            case $opt in
+                u)
+                    user="$OPTARG"
+                    ;;
+                p)
+                    password="$OPTARG"
+                    ;;
+                i)
+                    issuer_url="$OPTARG"
+                    ;;
+                c)
+                    client_id="$OPTARG"
+                    ;;
+                s)
+                    client_secret="$OPTARG"
+                    ;;
+                *)
+                    echo "Unknown option: -$OPTARG"
+                    usage
+                    ;;
+            esac
+        done
+}
+
+# This is needed to see if there is already a DocumentReference with the given project identifier.
+# In that case, the id of the DocumentReference is later used to update the DocumentReference.
+http_get_all_doc_refs() {
+    auth="$1"
+    if [ "$auth" == "no-auth" ]; then
+        response=$(curl -s "$report_server/DocumentReference" \
+                        -H "Content-Type: application/fhir+json")
+        echo "$response"
+    elif [ "$auth" == "basic-auth" ]; then
+        response=$(curl -s "$report_server/DocumentReference" \
+                        -H "Content-Type: application/fhir+json" \
+                        -u "$user:$password")
+        echo "$response"
+    else
+        oauth_response=$(curl -s -X POST "$issuer_url" \
+                            -H "Content-Type: application/x-www-form-urlencoded" \
+                            -d "grant_type=client_credentials" \
+                            -d "client_id=$client_id" \
+                            -d "client_secret=$client_secret")
+        fhir_report_bearer_token=$(echo "$oauth_response" | jq -r '.access_token')
+
+        response=$(curl -s "$report_server/DocumentReference" \
+                    -H "Content-Type: application/fhir+json" \
+                    -H "Authorization: Bearer $fhir_report_bearer_token")
+
+        echo "$response"
+    fi
+}
+
+get_doc_ref_id() {
+    auth="$1"
+    all_doc_refs=$(http_get_all_doc_refs "$auth")
+
+    matching_refs=$(echo "$all_doc_refs" | jq -c \
+    --arg sys "$proj_ident_sys" \
+    --arg val "$proj_ident_val" \
+    '.entry[]?.resource | select(.masterIdentifier.system == $sys and .masterIdentifier.value == $val)')
+
+    count=$(echo "$matching_refs" | jq -s 'length')
+
+    if [[ "$count" -gt 1 ]]; then
+        echo "Error: Multiple DocumentReferences exist for masterIdentifier {system: '$proj_ident_sys', value: '$proj_ident_val'}"
+        exit 1
+    elif [[ "$count" -eq 1 ]]; then
+        doc_id=$(echo "$matching_refs" | jq -r '.id')
+        echo "$doc_id"
+    else
+        echo ""
+    fi
+}
+
+create_bundle() {
+    auth="$1"
+    doc_ref_id=$(get_doc_ref_id "$auth")
+    err=$?
```

Comment:
Exit codes should be captured directly not via $?. Please ask an AI how to do that. I don't remember exactly.

Reply:
I mean this: https://github.com/koalaman/shellcheck/wiki/SC2181